src/main/java/io/cdap/plugin/date/DateTransformConfig.java

```diff
@@ -0,0 +1,290 @@
+        .withStacktrace(e.getStackTrace());
+    }
+
+    validateDateFormat(failureCollector, targetFormat, PROPERTY_TARGET_FORMAT);
+    validateDateFormat(failureCollector, sourceFormat, PROPERTY_SOURCE_FORMAT);
+
+    if (!containsMacro(PROPERTY_TARGET_FIELDS) && !containsMacro(PROPERTY_SOURCE_FIELDS)) {
+      if (this.getTargetFields().size() != this.getSourceFields().size()) {
+        failureCollector.addFailure("Target and source fields must contain the same number of fields.", null)
+          .withConfigProperty(PROPERTY_SOURCE_FIELDS)
+          .withConfigProperty(PROPERTY_TARGET_FIELDS);
+      }
+    }
+
+    if (inputSchema == null) {
+      failureCollector.addFailure("Input schema cannot be null.", null);
+      throw failureCollector.getOrThrowException();
+    }
+
+    List<String> sourceFields = getSourceFields();
+    for (String field : sourceFields) {
+      Schema.Field inputField = inputSchema.getField(field);
+      if (inputField == null) {
+        failureCollector.addFailure(String.format("Source field '%s' is not present in input schema.",
+                                                  field), null)
+          .withConfigProperty(PROPERTY_SOURCE_FIELDS)
+          .withInputSchemaField(field);
+      } else {
+        Schema inputFieldSchema = inputField.getSchema();
+
+        if (inputFieldSchema.isNullable()) {
+          inputFieldSchema = inputFieldSchema.getNonNullable();
+        }
+
+        if (inputFieldSchema.getType() != Schema.Type.STRING && inputFieldSchema.getType() != Schema.Type.LONG) {
+          failureCollector.addFailure(String.format("Source field '%s' must be of type 'string' or 'long'" +
+                                                      " but is of type '%s'.",
+                                                    inputField.getName(), inputFieldSchema.getDisplayName()),
+                                      null).withConfigProperty(PROPERTY_SOURCE_FIELDS)
+            .withInputSchemaField(field, null);
+        }
+      }
+    }
+  }
+
+  public void validateDateFormat(FailureCollector failureCollector, String value, String propertyName) {
+    if (!containsMacro(propertyName)) {
+      try {
+        new SimpleDateFormat(value);
+      } catch (IllegalArgumentException ex) {
+        failureCollector.addFailure(String.format("Field '%s' contains invalid date pattern '%s'",
+                                                  propertyName, value), "Use format specified in " +
+          "https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html")
+          .withConfigProperty(propertyName)
+          .withStacktrace(ex.getStackTrace());
+      }
+    }
+  }
+
+  @Nullable
+  public Schema getSchema() {
+    try {
+      return Strings.isNullOrEmpty(schema) ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to parse output schema: " + schema, e);
```

Comment:
This should be `IllegalArgumentException`

Reply:
Done